tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/provision/helm/wildfly/WildFlyHelmChartReleaseAdapter.java

```diff
@@ -0,0 +1,703 @@
+			Arrays.stream(adaptee.getBuild().getS2i().getGalleonLayers().split(","))
+					.forEach(fp -> result.add(fp));
+		}
+		return result;
+	}
+
+	@Override
+	public void setS2iGalleonLayers(LinkedHashSet<String> s2iGalleonLayers) {
+		if (adaptee.getBuild() == null) {
+			adaptee.setBuild(new Build());
+		}
+		if (adaptee.getBuild().getS2i() == null) {
+			adaptee.getBuild().setS2i(new S2i());
+		}
+		// let's check just here, so there's a way to clear, i.e. by passing an empty or null list
+		if ((s2iGalleonLayers != null) && !s2iGalleonLayers.isEmpty()) {
+			adaptee.getBuild().getS2i().setGalleonLayers(s2iGalleonLayers.stream().collect(Collectors.joining(",")));
+		}
+	}
+
+	@Override
+	public WildflyHelmChartRelease withS2iGalleonLayers(LinkedHashSet<String> s2iGalleonLayers) {
+		this.setS2iGalleonLayers(s2iGalleonLayers);
+		return this;
+	}
+
+	@Override
+	public WildflyHelmChartRelease withS2iGalleonLayers(String s2iGalleonLayers) {
+		LinkedHashSet s2iGalleonLayersSet = new LinkedHashSet<>(
+				Arrays.stream(s2iGalleonLayers.split(",")).collect(Collectors.toList()));
+		this.setS2iGalleonLayers(s2iGalleonLayersSet);
+		return this;
+	}
+
+	@Override
+	public WildflyHelmChartRelease withS2iGalleonLayer(String s2iGalleonLayer) {
+		if (adaptee.getBuild() == null) {
+			adaptee.setBuild(new Build());
+		}
+		if (adaptee.getBuild().getS2i() == null) {
+			adaptee.getBuild().setS2i(new S2i());
+		}
+		final List<String> galleonLayers = new ArrayList<>();
+		if (!Strings.isNullOrEmpty(adaptee.getBuild().getS2i().getGalleonLayers())) {
+			galleonLayers.addAll(Arrays.stream(adaptee.getBuild().getS2i().getGalleonLayers().split(","))
+					.collect(Collectors.toList()));
+		}
+		galleonLayers.add(s2iGalleonLayer);
+		adaptee.getBuild().getS2i().setGalleonLayers(galleonLayers.stream().collect(Collectors.joining(",")));
+		return this;
+	}
+
+	@Override
+	public LinkedHashSet<String> getS2iChannels() {
+		// no channels definition for WIldFly Helm Charts
+		return new LinkedHashSet<>();
+	}
+
+	@Override
+	public void setS2iChannels(LinkedHashSet<String> s2iChannels) {
+		log.warn(
+				"Setting s2i build channels on a WildFly release has no effect since WildFly Helm Charts don't support channels");
+	}
+
+	@Override
+	public WildflyHelmChartRelease withS2iChannels(LinkedHashSet<String> s2iChannels) {
+		this.setS2iChannels(s2iChannels);
+		return this;
+	}
+
+	@Override
+	public WildflyHelmChartRelease withS2iChannel(String s2iChannel) {
+		log.warn(
+				"Setting s2i build channels on a WildFly release has no effect since WildFly Helm Charts don't support channels");
+		return this;
+	}
+
+	@Override
+	public BuildMode getBuildMode() {
+		return adaptee.getBuild() == null ? null
+				: adaptee.getBuild().getS2i() != null ? BuildMode.S2I
+						: adaptee.getBuild().getBootableJar() != null ? BuildMode.BOOTABLE_JAR : null;
+	}
+
+	@Override
+	public String getBootableJarBuilderImage() {
+		return adaptee.getBuild() == null ? ""
```

Comment:
@fabiobrz  I see the counterpart method Eap8HelmChartReleaseAdapter.getBootableJarBuilderImage() doesn't return anything on purpose: 
```java
	@Override
	public String getBootableJarBuilderImage() {
		// no Bootable JAR image is supported by EAP 8 Helm Charts
		return "";
	}
```
did I understand correctly we will have to implement one more adapter for XP5?

Reply:
Well, it wouldn't seem the case to me, since the EAP Heml CHart should be valid for a bootable JAR build. That being said it actually depends on whether the developers would create yet another schema file.